src/Azure.DataApiBuilder.Mcp/Core/McpStdioServer.cs

```diff
@@ -0,0 +1,486 @@
+
+        /// <summary>
+        /// Handles the "initialize" JSON-RPC method by sending the MCP protocol version, server capabilities, and server info to the client.
+        /// </summary>
+        /// <param name="id">
+        /// The request identifier extracted from the incoming JSON-RPC request. Used to correlate the response with the request.
+        /// </param>
+        /// <remarks>
+        /// This method constructs and writes the MCP "initialize" response to STDOUT. It uses the protocol version defined by <c>PROTOCOL_VERSION</c>
+        /// and includes supported capabilities and server information. No notifications are sent here; the server waits for the client to send
+        /// "notifications/initialized" before sending any notifications.
+        /// </remarks>
+        private void HandleInitialize(JsonElement? id)
+        {
+            // Extract the actual id value from the request
+            object? requestId = id.HasValue ? GetIdValue(id.Value) : null;
+
+            // Create the initialize response
+            var response = new
+            {
+                jsonrpc = "2.0",
+                id = requestId,
+                result = new
+                {
+                    protocolVersion = _protocolVersion,
+                    capabilities = new
+                    {
+                        tools = new { listChanged = true },
+                        logging = new { }
+                    },
+                    serverInfo = new
+                    {
+                        name = "Data API Builder",
+                        version = "1.0.0"
+                    }
+                }
+            };
+
+            string json = JsonSerializer.Serialize(response);
+            Console.Out.WriteLine(json);
+        }
+
+        /// <summary>
+        /// Handles the "tools/list" JSON-RPC method by sending the list of available tools to the client.
+        /// </summary>
+        /// <param name="id">
+        /// The request identifier extracted from the incoming JSON-RPC request. Used to correlate the response with the request.
+        /// </param>
+        private void HandleListTools(JsonElement? id)
+        {
+            List<object> toolsWire = new();
+            int count = 0;
+
+            // Tools are expected to be registered during application startup only.
+            // If this ever changes and tools can be added/removed at runtime while
+            // requests are being handled, we may need to introduce locking here or
+            // have the registry return a thread-safe snapshot.
+            foreach (Tool tool in _toolRegistry.GetAllTools())
+            {
+                count++;
+                toolsWire.Add(new
+                {
+                    name = tool.Name,
+                    description = tool.Description,
+                    inputSchema = tool.InputSchema
+                });
+            }
+
+            WriteResult(id, new { tools = toolsWire });
```

Comment:
Why isnt the stdioRole required to be added to the httpcontext when listing the tools? Is that information not controlled by our authZ logic currently?